Snakefile

```diff
@@ -0,0 +1,260 @@
+    input:
+        rules.transform.output.sequences
+    output:
+        "data/gisaid/nextclade_rerun.tsv"
+    params:
+        batch_size = 1000
+    threads: 16
+    shell:
+        '''
+        ./bin/run-nextclade {params.batch_size} {threads} {input} {output}
+        '''
+
+rule join_metadata_and_clades:
+    input:
+        clades = rules.concat_clades.output.clades,
+        metadata = rules.transform.output.metadata,
+    output:
+        metadata = "data/{data_source}/new_metadata.tsv"
+    shell:
+        '''
+        ./bin/join-metadata-and-clades \
+            {input.metadata} \
+            {input.clades} \
+            -o {output.metadata}
+        '''
+
+rule flag_metadata:
+    input:
+        rules.join_metadata_and_clades.output.metadata
+    output:
+        "data/{data_source}/flagged_metadata.txt"
+    shell:
+        '''
+        ./bin/flag-metadata {input} > {output}
+        '''
+
+rule check_locations:
+    input:
+        rules.join_metadata_and_clades.output.metadata
+    output:
+        "data/{data_source}/location_hierarchy.tsv"
+    shell:
+        '''
+        ./bin/check-locations {input} {output} gisaid_epi_isl
+        '''
+
+rule clean:
+    message: "Removing directories: {params}"
+    params:
+        "data",
+        "tmp"
+    shell:
+        "rm -rfv {params}"
+
+rule metadata_addition:
+    input:
+        old = "data/{data_source}/inputs/metadata.tsv",
+        new = "data/{data_source}/new_metadata.tsv"
+    output:
+        additions = "data/{data_source}/metadata_additions.tsv",
+        changes = "data/{data_source}/metadata_changes.tsv"
+    params:
+        key = "gisaid_epi_isl"
+    shell:
+        '''
+        csv-diff \
+            {input.old} \
+            {input.new} \
+            --format tsv \
+            --key {params.key} \
+            --singular sequence \
+            --plural sequences \
+            > {output.changes} & \
+        ./bin/metadata-additions {input.old} {input.new} {params.key} >  {output.additions}
+        '''
+
+rule additional_info_change:
+    input:
+        old = "data/{data_source}/inputs/additional_info.tsv",
+        new = "data/{data_source}/additional_info.tsv"
+    output:
+        changes = "data/{data_source}/additional_info_changes.tsv"
+    params:
+        key = "gisaid_epi_isl"
+    shell:
+        '''
+        csv-diff \
+            <(awk 'BEGIN {{FS="\t"}}; {{ if ($3 != "" || $4 != "") {{ print }}}}' {input.old}) \
+            <(awk 'BEGIN {{FS="\t"}}; {{ if ($3 != "" || $4 != "") {{ print }}}}' {input.new}) \
+            --format tsv \
+            --key gisaid_epi_isl \
+            --singular "additional info" \
+            --plural "additional info" > {output.changes}
+        '''
+
+
+##########################################################################
+onerror:
+    shell("./bin/notify-slack 'ncov-ingest failed'")
+
+
+##########################################################################
+## target rules
```

Comment:
@rneher If remove these rules (locally) and this one:
https://github.com/nextstrain/ncov-ingest/pull/118/files#diff-47959dfd378b3cd1d39b5515418ee8e4444ab7a6036d5197a5bea82814f928a3R41

does that seem like a good way to be able to test running this on AWS since it won't mess with anything official by not uploading anything new to s3?

Seems like (?) we just need to make sure it has access to inputs and dependencies and then we can just do `nextstrain build --aws-batch`

Reply:
yes. or you wrap all pushed to S3 in an `echo "..."`. Then it prints them but doesn't do anything. These would be the rules `upload_gisaid_to_S3` and `push_gisaid_to_S3` and maybe `post_to_slack`. 

These rule names are a little convoluted, mostly since I didn't really know what would come further down the file...

better defining what needs updating after annotations get updated is also necessary. For example by splitting the `transform` into something that turns the gisaid json into metadata and sequences and another step that applies the annotations.